### ISSUE TITLE
Bugs in Docs: VictoryBar example causes TypeError

### ISSUE THREAD

**Issue by coopy** [FormidableLabs/victory-chart#133]
Repro:

1. `builder run docs-dev`
2. `open http://localhost:3000`
3. Scroll down to #Animating

See this in **Live Preview**:
```
TypeError: Array.prototype.fill called on null or undefined
```

If the `dataAttributes` prop of `<VictoryBar/>` is removed, then the example works. This could be a bug in `victory-core/victory-animation`:

```
        dataAttributes={[
          {fill: "cornflowerblue"},
          {fill: "tomato"},
          {fill: "orange"},
          {fill: "gold"}
        ]}
```

**Comment by divmain** [comment]
Did you track down where `Array.prototype.fill` was happening?  Unless it is being polyfilled, this method is not available in IE.

**Comment by coopy** [comment]
Nope, I did not dig deeper. This happened in Chrome.

**Comment by coopy** [comment]
@divmain I pushed a branch: https://github.com/FormidableLabs/victory-chart/tree/repro-133
It replaces the demo app with a reproduction of the problem.

You can `npm install && npm start`, and then open localhost:3000 for a reproduction.

**Comment by divmain** [comment]
Cross-posting from Slack:

In `app.jsx` there is a length mismatch between `this.state.data` (which is passed in on each `setState`) and the provided `dataAttributes` array.  In `getAttributes`, it checks for `dataAttributes[index]` which doesn't exist because of the length mismatch.  So it defaults to using the entire `dataAttributes` array for that iteration (because `dataAttributes` can also be a single object for all data).

Later, `attributes.fill` is `pick`ed when props are calculated.  This value is intended to be a color but instead is `Array.prototype.fill`.  Finally, this gets you to `evaluateProp` where `fill` is now a function, which is invoked.

It'd probably be worth throwing an error early if there is a length mismatch (or removing array-form `dataAttributes`), since this is a bit of a rabbit hole!

For the limited use case, this diff fixes the issue:
 ```diff
diff --git a/demo/app.jsx b/demo/app.jsx
index f81b69b..9eba591 100644
--- a/demo/app.jsx
+++ b/demo/app.jsx
@@ -48,7 +48,8 @@ class App extends React.Component {
           {fill: "cornflowerblue"},
           {fill: "tomato"},
           {fill: "orange"},
-          {fill: "gold"}
+          {fill: "gold"},
+          {fill: "red"}
         ]}
       />
     );
```

**Comment by divmain** [comment]
Closed by #138.

**Comment by coopy** [comment]
Thanks for your help, @divmain!
